src/ConfigOptions.xml

```diff
@@ -119,6 +119,10 @@
       <Value>before</Value>
     </Values>
   </Option>
+  <Option Id="ExpressionBodyStyleOnNextLine">
+    <ValuePlaceholder>true|false</ValuePlaceholder>
+    <Description>Place expression body always on next/same line</Description>
+  </Option>
```

Comment:
Could you please explain why did you add this option? I think that this option is basically the same as checking if analyzer is enabled or not. If you want to decide whether to wrap line before `=>` in RCS1016 (as an example) just check if RCS0062 is effective or not.

See extensions methods `IsEffective` for `DiagnosticDescriptor`.

Reply:
Basically you are right, but I wouldn't know how to use that information in the `CodeFixProvider`.
I found usings of `DiagnosticDescriptor.IsEffective` only in analyzers. The respective information is then unambiguous via the created diagnostic.

But configuring the behavior of a single diagnostic (RCS1016) in the `CodeFixProvider` is always done via `AnalyzerConfigOptions`.

Reply:
You can definitely get the information whether an analyzer is effective from `CodeFixProvider`. You need a `SyntaxTree`, which is property of a node and `CompilationOptions` which is a property of a `Project`:

```cs
DiagnosticRules.ConfigureAwait.IsEffective(
    node.SyntaxTree,
    context.Document.Project.CompilationOptions,
    cancellationToken);
```

Reply:
Thanks! I'm still making my way through to understand what to get from where.

So that would work, but now I would have dependency problems:
Currently, my attempt for the (not yet working) line breaking is in `Workspaces.Common.ConvertBlockBodyToExpressionBodyRefactoring`, but the `DiagnosticRules` are defined in `Formatting.Analyzers`.

To know, how to refactor that, I would need to know, how the actual new line is inserted. My approach doesn't work, yet.
Probably I would need to call the `SyntaxNode ConvertBlockBodyToExpressionBodyRefactoring.Refactor(SyntaxNode, AnalyzerConfigOptions)` from somewhere else, then manipulated the new syntax node, and finally replace the node in the document.

Reply:
Could you call `IsEffective` from code fix provider and then pass it to `ConvertBlockBodyToExpressionBodyRefactoring ` as a parameter? Something like `bool wrapLineBeforeArrowToken`?

Reply:
That also doesn't work, because the RCS0062 DiagnosticRule is defined in `Formatting.Analyzers`, but the caller of `ConvertBlockBodyToExpressionBodyRefactoring.RefactorAsync`, which would then have to evaluate the bool, like `UseBlockBodyOrExpressionBodyCodeFixProvider` (RCS1016) or `ConstructorDeclarationRefactoring` can't reference that project.

Reply:
@cbersch I moved `DiagnosticRules` and `DiagnosticIdentifiers` to `Roslynator.Common` so now it's accessible from everywhere. Please re-run `generate_code.ps1`.

Reply:
Grest, thanks. I'll rework when I'm back from vacaton